{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## AlphaGenome predictions data wrangling"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AlphaGenome predictions with RNA-seq modality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### modify this to convert the .tsv from AlphaGenome to .parquet for the model_benchmarking notebook. It was already done and the db directory\n",
    "### already has the output .parquet file in the required format, but I forgot to save the notebook, so lost the code. \n",
    "\n",
    "### To note: I adjusted the format to match the .parquet format needed for the AUC calculation and removed all rows with NAN values \n",
    "### The final df has ~17000 rows \n",
    "### Also create a version where all the NAN values are converted into zeros\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "alphagenome = pd.read_csv('/hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/results/predict_variant_kircher2019/result.alphagenome.variant.summary.tsv', sep='\\t')\n",
    "alphagenome[['chrom','cre_start','cre_end']] = alphagenome['Region'].str.split(r'[:\\-]', expand=True)\n",
    "alphagenome['Scored_Interval'] = alphagenome['Scored_Interval'].str.rstrip(r'[:\\.]')\n",
    "alphagenome['Scored_Interval'] = alphagenome['Scored_Interval'].fillna(alphagenome['Region'])\n",
    "\n",
    "alphagenome[['bin_chrom','bin_start','bin_end']] = alphagenome['Scored_Interval'].str.split(r'[:\\-]', expand=True)\n",
    "alphagenome['Raw_Score'] = pd.to_numeric(alphagenome['Raw_Score'], errors='coerce')\n",
    "mask_alt = alphagenome['Ref'] != alphagenome['Alt']\n",
    "alphagenome['log2FC'] = np.nan\n",
    "alphagenome.loc[mask_alt, 'log2FC'] = alphagenome.loc[mask_alt, 'Raw_Score']\n",
    "drop_mask = (alphagenome['Ref'] == alphagenome['Alt']) & (alphagenome['Raw_Score'] == 0)\n",
    "alphagenome = alphagenome.loc[~drop_mask].copy()\n",
    "alphagenome.reset_index(drop=True, inplace=True)\n",
    "\n",
    "\n",
    "alphagenome.rename(columns={'Position':'allele_pos',\n",
    "                            'Ref':'ref_allele',\n",
    "                            'Alt':'allele',\n",
    "                            'Raw_Score':'effect'}, inplace=True)\n",
    "alphagenome_v1 = alphagenome[['chrom','cre_start','cre_end','bin_chrom','bin_start','bin_end','allele_pos','ref_allele','allele','effect','log2FC']]\n",
    "alphagenome_v1.to_csv('/hpc/home/rv103/igvf/revathy/Blueplots/predictions/K562_AlphaGenome.txt', sep='\\t', index=False)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "min_col = alphagenome_v1['log2FC'].min()\n",
    "alphagenome_v1['effect'] = alphagenome_v1['effect'].fillna(min_col)\n",
    "alphagenome_v1['log2FC'] = alphagenome_v1['log2FC'].fillna(min_col)\n",
    "\n",
    "alphagenome_v1.to_parquet('/hpc/home/rv103/igvf/revathy/db/Kircher-2019_K562-AlphaGenome.parquet', engine='pyarrow', compression='snappy')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AlphaGenome with ATAC-seq modality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "alphagenome = pd.read_csv('/hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/results/predict_variant_kircher2019/result.alphagenome.K562.ATAC.2KB.variant.summary.tsv', sep='\\t')\n",
    "alphagenome[['chrom','cre_start','cre_end']] = alphagenome['Region'].str.split(r'[:\\-]', expand=True)\n",
    "alphagenome['Scored_Interval'] = alphagenome['Scored_Interval'].str.rstrip(r'[:\\.]')\n",
    "alphagenome['Scored_Interval'] = alphagenome['Scored_Interval'].fillna(alphagenome['Region'])\n",
    "\n",
    "alphagenome[['bin_chrom','bin_start','bin_end']] = alphagenome['Scored_Interval'].str.split(r'[:\\-]', expand=True)\n",
    "alphagenome['Raw_Score'] = pd.to_numeric(alphagenome['Raw_Score'], errors='coerce')\n",
    "mask_alt = alphagenome['Ref'] != alphagenome['Alt']\n",
    "alphagenome['log2FC'] = np.nan\n",
    "alphagenome.loc[mask_alt, 'log2FC'] = alphagenome.loc[mask_alt, 'Raw_Score']\n",
    "drop_mask = (alphagenome['Ref'] == alphagenome['Alt']) & (alphagenome['Raw_Score'] == 0)\n",
    "alphagenome = alphagenome.loc[~drop_mask].copy()\n",
    "alphagenome.reset_index(drop=True, inplace=True)\n",
    "\n",
    "\n",
    "alphagenome.rename(columns={'Position':'allele_pos',\n",
    "                            'Ref':'ref_allele',\n",
    "                            'Alt':'allele',\n",
    "                            'Raw_Score':'effect'}, inplace=True)\n",
    "alphagenome_v1 = alphagenome[['chrom','cre_start','cre_end','bin_chrom','bin_start','bin_end','allele_pos','ref_allele','allele','effect','log2FC']]\n",
    "alphagenome_v1.to_csv('/hpc/home/rv103/igvf/revathy/Blueplots/predictions/K562_AlphaGenome-ATACseq.txt', sep='\\t', index=False)\n",
    "\n",
    "# min_col = alphagenome_v1['log2FC'].min()\n",
    "# alphagenome_v1['effect'] = alphagenome_v1['effect'].fillna(min_col)\n",
    "# alphagenome_v1['log2FC'] = alphagenome_v1['log2FC'].fillna(min_col)\n",
    "\n",
    "alphagenome_v1.to_parquet('/hpc/home/rv103/igvf/revathy/db/Kircher-2019_K562-AlphaGenome-ATACseq.parquet', engine='pyarrow', compression='snappy')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "BlueSTARR",
   "language": "python",
   "name": "bluestarr"
  },
  "language_info": {
   "name": "python",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
